### PR TITLE
Python and TS: encode path params

### PR DESCRIPTION
**Pull request by joeldodge79**

### CODE REVIEW

File: src/sdkModels.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ class Property extends SchemadSymbol implements IProperty {
 
   constructor(name: string, type: IType, schema: OAS.SchemaObject, required: string[] = []) {
     super(name, type, schema)
-    this.required = required.includes(name)
+    this.required = !!(required.includes(name) || schema.required?.includes(name))
```
Comment (jkaster) [inline src/sdkModels.ts]: Thanks for this fix

File: typescript/looker/sdk/3.1/streams.ts [hunk 2]
```diff
@@ -2152,9 +2148,7 @@ export class Looker31SDKStream extends APIMethods {
     return this.authStream<IRenderTask>(
       callback,
       'POST',
-      encodeURI(
-        `/render_tasks/lookml_dashboards/${request.dashboard_id}/${request.result_format}`,
-      ),
+      `/render_tasks/lookml_dashboards/${request.dashboard_id}/${request.result_format}`,
```
Comment (jkaster) [inline typescript/looker/sdk/3.1/streams.ts]: if result_format is a string should it be getting url encoded?

File: python/looker_sdk/sdk/api31/methods.py [hunk 3]
```diff
@@ -279,6 +280,8 @@ def all_git_branches(
         transport_options: Optional[transport.PTransportSettings] = None,
     ) -> Sequence[models.GitBranch]:
         """Get All Git Branches"""
+        if project_id == urllib.parse.unquote(project_id):
+            project_id = urllib.parse.quote(project_id, safe="")
```
[the remark this answers is not in the document]
Comment (joeldodge79) [reply python/looker_sdk/sdk/api31/methods.py]: done

File: typescript/looker/sdk/3.1/streams.ts (hunk 2 above)
Comment (joeldodge79) [reply typescript/looker/sdk/3.1/streams.ts]: fixed

**Comment by joeldodge79** [comment]
@jkaster ready for you to take another look

**Pull request merged**